### ISSUE TITLE
Comments & Corrections

### ISSUE THREAD

**Issue by LukeHollingsworth** [TomGeorge1234/DeepLearningTutorial#1]
Hey Tom! Thanks for using me as a Guinea Pig for this, it was a lot of fun to go through. I have made a few comments below, mostly regarding typos and syntax corrections! 

- In the introduction: _"Coding a deep neural network by hand this **way** will help us gain an understanding of the mathematics going on behind the scenes."_
- Task set-up: super small syntax thing that might not be correct, seems like you want ith element of **x** and **y** instead of bold vectors
- Your drawing of the data stacked into matrices confused me, just by the various terms (N, N_cells, N_train etc.). Is **X** data flattened so N_data = N_timesteps * N_examples?
- Plot the data: _"The data: i**n**puts and outputs split into training (blue) and testing (green) fractions"_
- def fit(self):
        """Each model must have a fit function which ,in one way or another, uses the **t**raining data to optimise the parameters (weights) of the model."""
- Mention 'traindata=True' in plot_testing_performance( ), however it is never used
- In plot_training_error( ): not sure what you mean in the comment _"if comparison is not None, plot the error of those models onto for comparison"_
- Optional exercise 1.1: In solution, (I think) the penultimate line should be w*^T (?).
- 2.0 Deep Neural Networks: _"...solve the same task: **predicting** position from neural firing rates."_
- 2.2: Might be worth mentioning some of the pitfalls of gradient descent. You have a great little diagram that shows local (as well as global) minima and so it might help to mention the existence of these. This is also naturally related to learning rate which you have included above (mathematically) but not defined.
(- Just a positive note here: I absolutely love the colourful backprop maths <3 )
- 2.3.1: _"So we don't have to do an arduous `for`-loop over all the data point**s** (these are really slow in python)"_ and _"This means we can perform the sum over all the **data points** in a simple dot-product."_
- 2.3.1 Vectorising updates: I would replace lower-case weight w with upper-case W, as you have done for X, Y and N
- 2.7: _"where $C^{(n)}$ is define recursively **from** the layer after it:"_
- 3: "_...it intelligently works back through the **memory** trace to calculate the gradient..."_ and _"...so can be used to train big networks on big dat sets incredibly fast."_

P.S. This was a lot of fun to go through, I also learnt some new things along the way! I'm sure it will be very well-received when you present it to this summer school, they are lucky to have you there. I hope these comments help to polish it before then.

**Comment by TomGeorge1234** [comment]
cheers Luke for this valuable feedback!! I've implemented the changes
